inst/include/child_model_simulation.hpp

```diff
@@ -1003,25 +1017,25 @@ void hc_initiate_art_by_age(int t,
                             IntermediateData<ModelVariant, real_type> &intermediate) {
   static_assert(ModelVariant::run_child_model,
                 "hc_initiate_art_by_age can only be called for model variants where run_child_model is true");
-  constexpr auto ss = StateSpace<ModelVariant>().base;
-  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
-  const auto cpars = pars.children.children;
+  constexpr auto ss_b = StateSpace<ModelVariant>().base;
+  constexpr auto ss_c = StateSpace<ModelVariant>().children;
+  const auto& p_hc = pars.children.children;
+  auto& n_hc = state_next.children;
 
-  //all children under a certain age eligible for ART
-  for (int s = 0; s <ss.NS; ++s) {
-    for (int a = 0; a < cpars.hc_art_elig_age(t); ++a) {
-      for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-        for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-          if (a < hc_ss.hc2_agestart) {
-            state_next.children.hc_art_need_init(hd, cat, a, s) += state_next.children.hc1_hiv_pop(hd, cat, a, s);
-          } else if (hd < hc_ss.hc2DS) {
-            state_next.children.hc_art_need_init(hd, cat, a, s) += state_next.children.hc2_hiv_pop(hd, cat, a - hc_ss.hc2_agestart, s);
+  // all children under a certain age eligible for ART
+  for (int s = 0; s < ss_b.NS; ++s) {
+    for (int a = 0; a < p_hc.hc_art_elig_age(t); ++a) {
+      for (int cat = 0; cat < ss_c.hcTT; ++cat) {
+        for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+          if (a < ss_c.hc2_agestart) {
+            n_hc.hc_art_need_init(hd, cat, a, s) += n_hc.hc1_hiv_pop(hd, cat, a, s);
+          } else if (hd < ss_c.hc2DS) {
+            n_hc.hc_art_need_init(hd, cat, a, s) += n_hc.hc2_hiv_pop(hd, cat, a - ss_c.hc2_agestart, s);
```

Comment:
im seeing a lot of this pattern around
```cpp
if (a < ss_c.hc2_agestart) {
  // block 1
  tensor(..., a)
} else if (hd < ss_c.hc2DS) {
  // block 2
  tensor(..., a - ss_c.hc2_agestart)
}
```
around, i would like to sort that out but not sure of any nice way to do it, if all things in `block 2` used `a - ss_c.hc2_agestart` then we could just redefine the index and have one block of code but in `block 2` ive seen mixed usage of `a` and `a - ss_c.hc2_agestart`

Reply:
this is just based on a naive look at this code but cant there just be `hc_hiv_pop` instead of `hc1_hiv_pop` and `hc2_hiv_pop` and we just combine the age indices in the correct order? why are these different?

Reply:
oh i see they have a different number of disease stages? i think at the very least we can maybe overload the indexing function so that it just subtracts `ss_c.hc2_agestart` from age or something like that, ill keep thinking of any other ideas